File: Lab2/apz-pzpi-22-1-tokar-denys-lab2-code/src/models/Sensor.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const sensorSchema = new Schema({
    type: {
        type: String,
        required: true,
        enum: ['temperature', 'humidity', 'light', 'soil_moisture']
    },
    greenhouseId: { type: Schema.Types.ObjectId, ref: 'Greenhouse', required: true },
    model: { type: String, required: true },
    status: { type: String, enum: ['active', 'inactive'], default: 'active' },
    unit: { type: String, required: true },
    lastValue: { type: Schema.Types.Mixed },
    lastUpdated: { type: Date },
    isDefault: { type: Boolean, default: false }
});

module.exports = mongoose.model('Sensor', sensorSchema);

File: Lab4/apz-pzpi-22-1-tokar-denys-lab4-code/app/src/main/java/com/example/apz_pzpi_22_1_tokar_denys_lab4_code/data/model/Sensor.kt
package com.example.apz_pzpi_22_1_tokar_denys_lab4_code.data.model

data class SensorValue(
    val type: String, // 'temperature', 'humidity', 'light', 'soil_moisture'
    val model: String, // Унікальний ID сенсора, напр. "ESP32HWID_TEMP"
    val value: Any?,   // Може бути Double, Int, String залежно від типу
    val unit: String,
    val lastUpdated: String? // Дата як рядок, потім можна парсити
)